packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -805,12 +818,21 @@ class _PageViewState extends State<PageView> {
 
     return NotificationListener<ScrollNotification>(
       onNotification: (ScrollNotification notification) {
-        if (notification.depth == 0 && widget.onPageChanged != null && notification is ScrollUpdateNotification) {
-          final PageMetrics metrics = notification.metrics as PageMetrics;
-          final int currentPage = metrics.page.round();
-          if (currentPage != _lastReportedPage) {
-            _lastReportedPage = currentPage;
-            widget.onPageChanged(currentPage);
+        if (notification.depth == 0) {
```

Comment:
Is there a good reason why we need to add this to the framework? Couldn't you just listen to these notifications from your app? Or write a wrapper widget that does this for you?

Reply:
I can listen to notification in wrapper widget. I did this because there was already a onPageChanged callback and adding onScrollStarted and onScrollCancelled callback seems to follow the same pattern and others could also use it. Otherwise, how could other developers know PageView could dispatch the 2 other notications?